2022-11-15 13:18:54.225 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.launched (was None)
2022-11-15 13:18:54.227 | INFO     | sc2.controller:create_game:37 - Creating new game
2022-11-15 13:18:54.228 | INFO     | sc2.controller:create_game:38 - Map:     OxideAIE
2022-11-15 13:18:54.230 | INFO     | sc2.controller:create_game:39 - Players: Bot Turtles(Terran), Computer VeryHard(Random, RandomBuild)
2022-11-15 13:18:54.232 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.init_game (was Status.launched)
2022-11-15 13:19:03.405 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.in_game (was None)
2022-11-15 13:19:03.407 | INFO     | sc2.main:_play_game:221 - Player 1 - Bot Turtles(Terran)
2022-11-15 13:19:04.073 | 00:00    0ms    0M    0G  12/ 15U [Version] 5.0.10.88500
2022-11-15 13:19:04.074 | 00:00    0ms    0M    0G  12/ 15U [Version] 5.0.6 Balance mod recognized.
2022-11-15 13:19:04.141 | 00:00    0ms    0M    0G  12/ 15U [ZoneManager] Map set to: MapName.Oxide from name: Oxide AIE and hash: 3281780.
2022-11-15 13:19:05.008 | ERROR    | sc2.main:initialize_first_step:140 - Caught unknown exception in AI on_start: Cannot instantiate typing.Any
Traceback (most recent call last):

  File "C:\Users\sword\OneDrive\Desktop\27turtles\run.py", line 18, in <module>
    main()
    └ <function main at 0x000001E5B0F02A70>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\run.py", line 13, in main
    result, opponentid = run_ladder_game(bot)
                         │               └ <sc2.player.Bot object at 0x000001E5B0F0B670>
                         └ <function run_ladder_game at 0x000001E58E6C9360>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\ladder.py", line 39, in run_ladder_game
    return stand_alone_game(bot), None
           │                └ <sc2.player.Bot object at 0x000001E5B0F0B670>
           └ <function stand_alone_game at 0x000001E5AE6095A0>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\ladder.py", line 168, in stand_alone_game
    return sc2.main.run_game(sc2.maps.get(map_name), [bot, Computer(Race.Random, Difficulty.VeryHard)], realtime=False,)
           │   │    │        │   │    │   │           │    │        │    │       │          └ <Difficulty.VeryHard: 7>
           │   │    │        │   │    │   │           │    │        │    │       └ <enum 'Difficulty'>
           │   │    │        │   │    │   │           │    │        │    └ <Race.Random: 4>
           │   │    │        │   │    │   │           │    │        └ <enum 'Race'>
           │   │    │        │   │    │   │           │    └ <class 'sc2.player.Computer'>
           │   │    │        │   │    │   │           └ <sc2.player.Bot object at 0x000001E5B0F0B670>
           │   │    │        │   │    │   └ 'OxideAIE'
           │   │    │        │   │    └ <function get at 0x000001E5AE6B84C0>
           │   │    │        │   └ <module 'sc2.maps' from 'C:\\Users\\sword\\OneDrive\\Desktop\\27turtles\\sc2\\maps.py'>
           │   │    │        └ <module 'sc2' from 'C:\\Users\\sword\\OneDrive\\Desktop\\27turtles\\sc2\\__init__.py'>
           │   │    └ <function run_game at 0x000001E5AE798DC0>
           │   └ <module 'sc2.main' from 'C:\\Users\\sword\\OneDrive\\Desktop\\27turtles\\sc2\\main.py'>
           └ <module 'sc2' from 'C:\\Users\\sword\\OneDrive\\Desktop\\27turtles\\sc2\\__init__.py'>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sc2\main.py", line 496, in run_game
    result: Result = asyncio.run(_host_game(map_settings, players, **kwargs))
                     │       │   │          │             │          └ {'realtime': False}
                     │       │   │          │             └ [<sc2.player.Bot object at 0x000001E5B0F0B670>, <sc2.player.Computer object at 0x000001E5B10585B0>]
                     │       │   │          └ Map(C:\Program Files (x86)\StarCraft II\maps\OxideAIE.SC2Map)
                     │       │   └ <function _host_game at 0x000001E5AE7989D0>
                     │       └ <function run at 0x000001E59037D7E0>
                     └ <module 'asyncio' from 'C:\\Users\\sword\\AppData\\Local\\Programs\\Python\\Python310\\lib\\asyncio\\__init__.py'>

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
           │    │                  └ <coroutine object _host_game at 0x000001E5B105C5F0>
           │    └ <function BaseEventLoop.run_until_complete at 0x000001E59037F1C0>
           └ <ProactorEventLoop running=True closed=False debug=False>

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 633, in run_until_complete
    self.run_forever()
    │    └ <function ProactorEventLoop.run_forever at 0x000001E59042BBE0>
    └ <ProactorEventLoop running=True closed=False debug=False>

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 600, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x000001E590384CA0>
    └ <ProactorEventLoop running=True closed=False debug=False>

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1896, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x000001E5902EC280>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sc2\main.py", line 369, in _host_game
    result = await _play_game(players[0], client, realtime, portconfig, game_time_limit, rgb_render_config)
                   │          │           │       │         │           │                └ None
                   │          │           │       │         │           └ None
                   │          │           │       │         └ None
                   │          │           │       └ False
                   │          │           └ <sc2.client.Client object at 0x000001E5B10594B0>
                   │          └ [<sc2.player.Bot object at 0x000001E5B0F0B670>, <sc2.player.Computer object at 0x000001E5B10585B0>]
                   └ <function _play_game at 0x000001E5AE798820>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sc2\main.py", line 226, in _play_game
    result = await _play_game_ai(client, player_id, player.ai, realtime, game_time_limit)
                   │             │       │          │      │   │         └ None
                   │             │       │          │      │   └ False
                   │             │       │          │      └ <turtles.bot.Turtles object at 0x000001E590376C80>
                   │             │       │          └ <sc2.player.Bot object at 0x000001E5B0F0B670>
                   │             │       └ 1
                   │             └ <sc2.client.Client object at 0x000001E5B10594B0>
                   └ <function _play_game_ai at 0x000001E5AE798670>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sc2\main.py", line 145, in _play_game_ai
    result = await initialize_first_step()
                   └ <function _play_game_ai.<locals>.initialize_first_step at 0x000001E5B0F037F0>

> File "C:\Users\sword\OneDrive\Desktop\27turtles\sc2\main.py", line 136, in initialize_first_step
    await ai.on_start()
          │  └ <function KnowledgeBot.on_start at 0x000001E5B0BB6CB0>
          └ <turtles.bot.Turtles object at 0x000001E590376C80>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sharpy\knowledges\knowledge_bot.py", line 71, in on_start
    await self.knowledge.start()
          │    │         └ <function Knowledge.start at 0x000001E5AE81B010>
          │    └ <sharpy.knowledges.knowledge.Knowledge object at 0x000001E58E356590>
          └ <turtles.bot.Turtles object at 0x000001E590376C80>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sharpy\knowledges\knowledge.py", line 151, in start
    await manager.post_start()
          │       └ <function ActManager.post_start at 0x000001E5AE684F70>
          └ <sharpy.managers.core.act_manager.ActManager object at 0x000001E5B1153EE0>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\sharpy\managers\core\act_manager.py", line 25, in post_start
    self._act = await self._act_or_func()
    │                 │    └ <bound method Turtles.create_plan of <turtles.bot.Turtles object at 0x000001E590376C80>>
    │                 └ <sharpy.managers.core.act_manager.ActManager object at 0x000001E5B1153EE0>
    └ <sharpy.managers.core.act_manager.ActManager object at 0x000001E5B1153EE0>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\turtles\bot.py", line 53, in create_plan
    self.build = ReactiveBio()
    │    │       └ <class 'turtles.reactivebio.ReactiveBio'>
    │    └ <function BotAI.build at 0x000001E5AE58BE20>
    └ <turtles.bot.Turtles object at 0x000001E590376C80>

  File "C:\Users\sword\OneDrive\Desktop\27turtles\turtles\reactivebio.py", line 95, in __init__
    skip_until=Any(
               └ typing.Any

  File "C:\Users\sword\AppData\Local\Programs\Python\Python310\lib\typing.py", line 387, in __call__
    raise TypeError(f"Cannot instantiate {self!r}")

TypeError: Cannot instantiate typing.Any
2022-11-15 13:19:05.107 | ERROR    | sc2.main:initialize_first_step:141 - Resigning due to previous error
2022-11-15 13:19:05.111 | Result: Defeat
2022-11-15 13:19:05.112 | Duration: 00:00
2022-11-15 13:19:05.112 | Step time avg: 0
2022-11-15 13:19:05.114 | Step time max: 0
2022-11-15 13:19:05.115 | [LostUnitsContents] My lost units minerals and gas: (0, 0)
2022-11-15 13:19:05.117 | [LostUnitsContents] My lost units minerals by score: 0.0
2022-11-15 13:19:05.118 | [LostUnitsContents] Enemy lost units minerals and gas: (0, 0)
2022-11-15 13:19:05.118 | [LostUnitsContents] Enemy lost units minerals by score: 0.0
2022-11-15 13:19:05.119 | [GameAnalyzerEnd] Own units:
2022-11-15 13:19:05.120 | [GameAnalyzerEnd] Enemy units:
2022-11-15 13:19:05.121 | [GameAnalyzerEnd] Minerals max 0 Average 0
2022-11-15 13:19:05.121 | [GameAnalyzerEnd] Vespene max 0 Average 0
2022-11-15 13:19:05.122 | INFO     | sc2.main:_play_game:228 - Result for player 1 - Bot Turtles(Terran): Defeat
2022-11-15 13:19:06.104 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.launched (was Status.in_game)
2022-11-15 13:19:06.110 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.quit (was Status.launched)
2022-11-15 13:19:06.111 | INFO     | sc2.sc2process:_close_connection:231 - Closing connection at 53534...
2022-11-15 13:19:06.113 | INFO     | sc2.sc2process:kill_all:33 - kill_switch: Process cleanup for 1 processes
